components/notifier/src/quality_time_notifier.py

```diff
@@ -23,18 +23,32 @@ async def notify(log_level: int = None) -> None:
             response = await session.get(f"http://{server_host}:{server_port}/api/v3/reports")
             json = await response.json()
 
-        notifications = reds_per_report(json)
+        notifications = reds_that_are_new(json)
 
+        total_new_red_metrics = 0
         for notification in notifications:
-            text = "\n\r[" + notification["report_title"] + "](" + notification["url"] + \
-                   ") contains " + str(notification["red_metrics"]) + " red metrics"
-            send = send_notification_to_teams(notification["teams_webhook"], text)
-            if not send:
-                logging.warning("unable to send the notification for %s", notification["report_uuid"])
+            total_new_red_metrics += notification["new_red_metrics"]
+        if total_new_red_metrics > 0:
+            for notification in notifications:
+                text = build_notification_text(notifications)
```

Comment:
Why are you passing all notifications to create the notification for a specific report? If report A has a web hook it means it also receives the red metrics for report B, right? That doesn't seem correct.

Reply:
corrected the message to only contain 1 report, and removed the leading sentence as it's now unnecessary